### PR TITLE
Feature 948/override slug validation

### LINKED ISSUE #948: Add toggle to indicate that a user is trying to migrate content types into CPTUI
Right now we prevent saving of content types where the slug is already registered elsewhere, whether it be a different post type, page slug, etc.

Let's offer a checkbox that essentially indicates "I'm trying to migrate things in to CPTUI, let me save this". 

The checked value should only be used to allow saving anyway, and not reject. I believe we have a filter on this, but I'd need to double check.

The checkbox should not save a value anywhere, so we do not have a new option to save. Just needs to be used temporarily on save.

### PR DESCRIPTION
**Pull request by ramsesdelr**
- Added a checkbox to bypass slug Frontend validation.
- Added a function to the `cptui_post_type_slug_exists ` filter to bypass slug validation in the backend.

### CODE REVIEW

File: build/cptui-scripts.js [hunk 1]
```diff
@@ -79,8 +79,9 @@ postboxes.add_postbox_toggles(pagenow);
             }
         }
         var $slugexists = $("#slugexists");
+        var $overwide_validation = $("#override_validation").is(":checked");
         if (typeof cptui_type_data != "undefined") {
-            if (cptui_type_data.existing_post_types.hasOwnProperty(value) && value !== original_slug) {
+            if (cptui_type_data.existing_post_types.hasOwnProperty(value) && value !== original_slug && $overwide_validation == false) {
```
Comment (tw2113) [inline build/cptui-scripts.js:84]: I think you mean override, not overwide.

File: src/js/cptui-scripts.js [hunk 2]
```diff
@@ -103,9 +103,10 @@ postboxes.add_postbox_toggles(pagenow);
 			}
 		}
 
-		var $slugexists = $('#slugexists');
+		var $slugexists          = $('#slugexists');
+		var $overwide_validation = $('#override_validation').is(":checked");
 		if ( typeof cptui_type_data != 'undefined' ) {
-			if (cptui_type_data.existing_post_types.hasOwnProperty(value) && value !== original_slug) {
+			if (cptui_type_data.existing_post_types.hasOwnProperty(value) && value !== original_slug && $overwide_validation == false ) {
```
Comment (tw2113) [inline src/js/cptui-scripts.js:109]: Same thing, but this time I'm commenting in the src version, not the build :D Sorry about that.

File: inc/post-types.php [hunk 3]
```diff
@@ -2444,6 +2449,24 @@ function cptui_updated_post_type_slug_exists( $slug_exists, $post_type_slug = ''
 }
 add_filter( 'cptui_post_type_slug_exists', 'cptui_updated_post_type_slug_exists', 11, 3 );
 
+/**
+ * Ignores the slug validation for an existing CPT if the override checkbox was previously selected.
+ *
+ * @since 1.14.0
```
Comment (tw2113) [inline inc/post-types.php:2455]: can you change this to "NEXT" or similar? I pushed out 1.14.0 earlier this morning.

File: inc/post-types.php [hunk 4]
```diff
@@ -330,8 +330,13 @@ function cptui_manage_post_types() {
 							<a href="#" id="auto-clear"><?php esc_html_e( 'Clear labels', 'custom-post-type-ui' ); ?></a>
 								<?php
 							echo $ui->get_td_end() . $ui->get_tr_end(); // phpcs:ignore.
-
+							echo $ui->get_tr_start() . $ui->get_th_start() . esc_html__( 'I\'m trying to migrate things in to CPTUI, let me save this', 'custom-post-type-ui' ) . $ui->get_th_end(); // phpcs:ignore.
+							echo $ui->get_td_start(); // phpcs:ignore.
 								?>
+								<input type="checkbox" name="cpt_override_validation" value=1 id="override_validation" />
```
Comment (tw2113) [inline inc/post-types.php:336]: Nitpicky, but let's get the value attribute wrapped in quotes

**Comment by tw2113** [comment]
Let's add in a little more touch. Only show the field on the "add new" tab. We don't need to make use of it when editing an entry. There's the `&action=edit` URL parameter that can be used here.

**Comment by ramsesdelr** [comment]
Thanks @tw2113  I applied the requested changes based on your feedback and also added the validation to display the checkbox only on the "Add New Post Type" page.

**Comment by tw2113** [comment]
Moved to master branch for the PR at the moment, as I need to start planning out followup release.

I'll resolve the conflicts at that time as well.

**Review by tw2113 — APPROVED** [review]

**Review by tw2113 — APPROVED** [review]

**Pull request merged**